docs/recipes/git-show.rst

```diff
@@ -31,7 +31,7 @@ Show SHA hash
 Show diff
 ======================================================================
 
-    >>> diff = commit.tree.diff()
+    >>> diff = commit.parents[0].tree.diff_to_tree(commit.tree)
```

Comment:
I think we should prefer to recommend `repo.diff(commit.parents[0], commit)` here, as that accepts treeishes in a few forms.